src/rust/fgbutil/src/main.rs

```diff
@@ -34,13 +38,54 @@ enum Format {
     Geojson,
 }
 
+// TODO: Cannot pass around GeozeroDatasource because it is a non object safe trait?
+// TODO: Regardless might need dyn / Box?
+
+fn write(format: Format, reader: GeozeroDatasource, output: BufWriter<File>) -> Result<()> {
+    match format {
+        Format::Geojson => write_flatgeobuf(reader, output)?,
+        Format::Flatgeobuf => write_flatgeobuf(reader, output)?
+    }
+    Ok(())
+}
+
+fn write_geojson(reader: GeozeroDatasource, output: BufWriter<File>) -> Result<()> {
```

Comment:
`reader: GeozeroDatasource` should be `mut reader: impl GeozeroDatasource`

Reply:
Thanks @pka, it does work and I now know a little more Rust. :)